tests/tests.js

```diff
@@ -0,0 +1,259 @@
+    assert.equal(state.startDateString, "20170214", "Checking startDateString");
+    assert.equal(state.endDateString, "20170220", "Checking endDateString");
+    // What kind of assertion for uniqueId?
+    assert.deepEqual(state.checkedDays, {});
+    
+    //assert that an exception is thrown with invalid dates.
+    var params = {startDate: "2017-02-14", endDate: "2017-01-01", calendarTitle: "hello"};
+    // I don't understand the qunit api for assert.raises (or assert.throws)
+    assert.raises(function() {
+        emptyCalendarState(params);
+        }, Error, "testing invalid dates");
+    assert.raises(function() {
+        emptyCalendarState(params);
+    }, /End date must be a date after start date./, "Testing error message");
+});
+
+// Testing calendar object 
+QUnit.module( "Calendar Tests", {
+
+beforeEach: function() {
+    // Prepare something once for all tests
+    this.params = {startDate: "2017-02-14" , endDate: "2017-02-20" , calendarTitle: "Test Calendar"};
+    //console.log(this.params);
+    this.state = emptyCalendarState(this.params);
+    
+    this.calendar = new Calendar(this.state);
+    
+}
+
+});
+
+QUnit.test("Init calendar object test", function(assert) {
+    assert.expect(7);
+  
+    assert.deepEqual(this.calendar.state, this.state, "Check state is correct");
+    assert.ok(moment.isMoment(this.calendar.startDate, "startDate is a moment obj")); 
+    assert.ok(moment.isMoment(this.calendar.endDate, "endDate is a moment obj"));
+    assert.equal(this.calendar.startDate.format("YYYYMMDD"), "20170214", "Checking startDate object's date");
+    assert.equal(this.calendar.endDate.format("YYYYMMDD"), "20170220", "Checking endDate object's date");
+    assert.equal(this.calendar.monthObjects.length, 1);
+    assert.equal(typeof(this.calendar.monthObjects[0]), "object", "Month object is object");
+});
+
+
+
+QUnit.test("addMonth calendar method test", function(assert) {
+    assert.expect(4);
+    this.calendar.addMonth();
+    
+    assert.equal(this.calendar.endDate.format("YYYYMMDD"), "20170320", "Check one month was added to endDate object");
+    assert.equal(this.calendar.state.endDateString, "20170320", "Check calendar endDateString added a month");
+    assert.equal(this.calendar.monthObjects.length, 2, "Check month was added to monthObjects array");
+    assert.equal(typeof(this.calendar.addMonth()), "object", "addMonth returns an object");
+});
+
+QUnit.test("generateMonthObjects calendar method test", function(assert) {
+    
+    assert.expect(7);
+    
+    var monthObjects = this.calendar.generateMonthObjects(moment("20170101", "YYYYMMDD"), moment("20170214", "YYYYMMDD"));
+
+    assert.ok(Array.isArray(monthObjects));
+    assert.equal(monthObjects[0].dateString, "20170101");
+    assert.equal(monthObjects[1].dateString, "20170201");
+    assert.equal(monthObjects[1].lastActiveDay, 14);
+    assert.equal(monthObjects.length, 2);
+    // Don't understand why this assertion passes
```

Comment:
These assertions check to make sure that your function fails as expected. The assertion succeeds because the function failed in the way you described.